src/python/pants/backend/python/mixed_interpreter_constraints/py_constraints.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+from collections import defaultdict
+
+from pants.backend.python.util_rules.pex import PexInterpreterConstraints
+from pants.engine.addresses import Addresses
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.rules import Get, collect_rules, goal_rule
+from pants.engine.target import TransitiveTargets, TransitiveTargetsRequest
+from pants.python.python_setup import PythonSetup
+
+logger = logging.getLogger(__name__)
+
+
+class PyConstraintsSubsystem(GoalSubsystem):
+    """Determine what Python interpreter constraints are used by files/targets."""
+
+    name = "py-constraints"
```

Comment:
This is meant to align with the field and option name `interpreter_constraints`.

Please feel free to bikeshed the name. Changing a goal name is costly.